pkg/repo/badgerdb/translation.go

```diff
@@ -22,25 +22,28 @@ func translationKey(serviceID uuid.UUID, language language.Tag) []byte {
 
 // SaveTranslation handles both Create and Update.
 func (r *Repo) SaveTranslation(ctx context.Context, serviceID uuid.UUID, translation *model.Translation) error {
+	if r.tx != nil { // use existing tx
+		return r.saveTranslation(ctx, serviceID, translation)
+	}
+
+	return r.Tx(ctx, func(ctx context.Context, rp repo.Repo) error { // create new tx
+		return rp.SaveTranslation(ctx, serviceID, translation) //nolint:wrapcheck
+	})
```

Comment:
Create a private and reusable helper function to ensure we have a tx. It helps to reduce boilerplate code when implementing Repo methods that require tx.

```suggestion
	return r.ensureTx(ctx, func(ctx context.Context, rp repo.Repo) error { // create new tx
		return rp.SaveTranslation(ctx, serviceID, translation) //nolint:wrapcheck
	})
```

Reply:
Please note, that we don't need `saveTranslation` anymore.